src/ThemesCleanDrupal8.php

```diff
@@ -0,0 +1,138 @@
+     *   digipolis.themes.drupal8 config value.
+     * @param array $dirs
+     *   The directories in which to search for the themes. Defaults to the
+     *   digipolis.root.project and digipolis.root.web config values, or the
+     *   current working directory if that is not set.
+     */
+    public function __construct($themes = [], $dirs = null)
+    {
+        $this->themes = $themes;
+        $this->dirs = $dirs;
+        $this->finder = new Finder();
+    }
+
+    /**
+     * Sets the themes to clean.
+     *
+     * @param array $themes
+     *   An array of Drupal theme machine names. Defaults to the keys of the
+     *   digipolis.themes.drupal8 config value.
+     */
+    public function themes($themes)
+    {
+        $this->themes = $themes;
+    }
+
+    /**
+     * Sets the directories in which to search for themes.
+     *
+     * @param array $dirs
+     *   The directories in which to search for the themes.
+     */
+    public function dirs($dirs)
+    {
+        $this->dirs = $dirs;
+    }
+
+    /**
+     * Sets the finder.
+     *
+     * @param \Symfony\Component\Finder\Finder $finder
+     *
+     * @return $this
+     *
+     * @codeCoverageIgnore
+     */
+    public function finder(Finder $finder)
+    {
+        $this->finder = $finder;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run()
+    {
+        $themesFromConfig = $this->getConfig()->get('digipolis.themes.drupal8', false);
+        if ($themesFromConfig) {
+            $themesFromConfig = array_keys((array) $themesFromConfig);
+        }
+        $themes = empty($this->themes)
+            ? $themesFromConfig
+            : $this->themes;
+        $dirsFromConfig = array_filter(
+            [
+                $this->getConfig()->get('digipolis.root.project', false),
+                $this->getConfig()->get('digipolis.root.web', false),
+            ]
+        );
+        $dirs = empty($this->dirs)
+            ? $dirsFromConfig
+            : $this->dirs;
+        if (empty($dirs)) {
+            $dirs = [getcwd()];
+        }
+
+        $finder = clone $this->finder;
+        $finder->in($dirs)->files();
+        foreach ($themes as $themeName) {
+            // Matches 'themes/(custom/){randomfoldername}/{themename}.info.yml'.
+            $finder->path('/themes\/(custom\/)?[^\/]*\/' . preg_quote($themeName, '/') . '\.info\.yml/');
+        }
+        $processed = [];
+        $collection = $this->collectionBuilder();
+        foreach ($finder as $infoFile) {
+            $path = dirname($infoFile->getRealPath());
+            // The web dir can be a subdir of the project root (in most cases
+            // really). Make sure we don't clean the same theme twice.
+            if (isset($processed[$path])) {
+                continue;
+            }
+            $processed[$path] = true;
+            $collection->addTask($this->taskThemeClean($path));
```

Comment:
We could if the theme & compile tasks implement CommandInterface. That way the compile and clean tasks can be converted to a single command (which basically means that all parallel tasks are concatenated with &&). Those commands will then be executed in parallel. But AFAIK commands concatenated with && are not executed in parallel but sequential... So it's a tradeoff between doing the actual compile and clean of multiple themes in parallel and the individual tasks for clean and compile sequentially, or the other way round: compile & clean themes sequentially and doing the individual tasks in paralell.